File: library/src/main/java/com/yy/codex/uikit/CAShapeLayer.java
package com.yy.codex.uikit;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Path;
import android.support.annotation.NonNull;

/**
 * Created by adi on 17/1/10.
 */

public class CAShapeLayer extends CALayer {

    private Path mPath = null;
    private int mStrokeColor = Color.BLACK;
    private double mLineWidth = 1;

    /* category CAShapeLayer Constructor */

    @Override
    protected void drawInCanvas(@NonNull Canvas canvas) {
        super.drawInCanvas(canvas);

        if (mPath != null){
            Paint paint = new Paint(Paint.ANTI_ALIAS_FLAG);
            paint.setStyle(Paint.Style.STROKE);
            paint.setStrokeWidth((float) mLineWidth);
            paint.setColor(mStrokeColor);
            canvas.drawPath(mPath, paint);
        }
    }

    /* category CAShapeLayer Getter&Setter */

    public Path getPath() {
        return mPath;
    }

    public int getStrokeColor() {
        return mStrokeColor;
    }

    public double getLineWidth() {
        return mLineWidth;
    }

    public @NonNull CAShapeLayer setPath(Path mPath) {
        this.mPath = mPath;
        return this;
    }

    public @NonNull CAShapeLayer setStrokeColor(int mStrokeColor) {
        this.mStrokeColor = mStrokeColor;
        return this;
    }

    public CAShapeLayer setLineWidth(double mLineWidth) {
        this.mLineWidth = mLineWidth;
        return this;
    }
}

File: library/src/main/java/com/yy/codex/uikit/CAShapeLayer.kt
package com.yy.codex.uikit

import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.Path

/**
 * Created by adi on 17/1/10.
 */

class CAShapeLayer : CALayer() {

    var path: Path? = null
        set(value) {
            if (field != value) {
                field = value
                this.setNeedDisplay(true)
            }
        }

    var strokeColor = Color.BLACK
        set(value) {
            if (field != value) {
                field = value
                this.setNeedDisplay(true)
            }
        }

    var lineWidth = 1.0
        set(value) {
            if (field != value) {
                field = value
                this.setNeedDisplay(true)
            }
        }

    /* category CAShapeLayer Constructor */

    override fun drawInCanvas(canvas: Canvas) {
        super.drawInCanvas(canvas)
        path?.let {
            val paint = Paint(Paint.ANTI_ALIAS_FLAG)
            paint.style = Paint.Style.STROKE
            paint.strokeWidth = lineWidth.toFloat()
            paint.color = strokeColor
            canvas.drawPath(it, paint)
        }
    }

}
